AutoCollection/diagnostic-channel/SpanParser.ts

```diff
@@ -17,22 +17,19 @@ export function spanToTelemetryContract(span: Span): (Contracts.DependencyTeleme
     const id = `|${span.spanContext().traceId}.${span.spanContext().spanId}.`;
     const duration = Math.round(span["_duration"][0] * 1e3 + span["_duration"][1] / 1e6);
     let peerAddress = span.attributes["peer.address"] ? span.attributes["peer.address"].toString() : "";
-    let component = span.attributes["component"] ? span.attributes["component"].toString() : "";
 
-    const isHttp: boolean = ((component).toUpperCase() === Constants.DependencyTypeName.Http) || (!!span.attributes[Constants.SpanAttribute.HttpUrl]);
-    const isGrpc: boolean = (component).toLowerCase() === Constants.DependencyTypeName.Grpc;
+    const isHttp: boolean = (!!span.attributes[Constants.SpanAttribute.HttpStatusCode]) || (!!span.attributes[Constants.SpanAttribute.HttpUrl]);
+    const isGrpc: boolean = (!!span.attributes[Constants.SpanAttribute.GrpcStatusCode]);
     if (isHttp) {
         // Read http span attributes
         const method = span.attributes[Constants.SpanAttribute.HttpMethod] || "GET";
         const url = new URL(span.attributes[Constants.SpanAttribute.HttpUrl].toString());
-        const host = span.attributes[Constants.SpanAttribute.HttpHost] || url.host;
-        const port = span.attributes[Constants.SpanAttribute.HttpPort] || url.port || null;
         const pathname = url.pathname || "/";
 
         // Translate to AI Dependency format
         const name = `${method} ${pathname}`;
         const dependencyTypeName = Constants.DependencyTypeName.Http;
-        const target = port ? `${host}:${port}`.toString() : host.toString();
+        const target = span.attributes[Constants.SpanAttribute.HttpUrl].toString() || undefined;
```

Comment:
- for HTTP, target is `[new URL(span[HTTP_URL])].hostname` [reference](https://github.com/Azure/azure-sdk-for-js/blob/df4ad49a92134f94540b43ed34bbe5605eb1164a/sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts#L127)
- for GRPC, target is` span[GRPC_METHOD]` [reference](https://github.com/Azure/azure-sdk-for-js/blob/df4ad49a92134f94540b43ed34bbe5605eb1164a/sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts#L121)
- else, target is `span[HTTP_URL] | undefined` [reference](https://github.com/Azure/azure-sdk-for-js/blob/df4ad49a92134f94540b43ed34bbe5605eb1164a/sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts#L104)